src/common/abstractPromote.ts

```diff
@@ -48,20 +59,66 @@ export abstract class PromoteCommand<T extends typeof SfCommand> extends SfComma
 
   protected async executePromotion(): Promise<PromotePipelineResult> {
     validateTestFlags(this.flags['test-level'], this.flags.tests);
-    const doceOrg: Org = await Org.create({ aliasOrUsername: this.flags['devops-center-username']?.getUsername() });
-    const pipelineStage: PipelineStage = await fetchAndValidatePipelineStage(
+    const doceOrg: Org = this.flags['devops-center-username'] as Org;
```

Comment:
@EstebanRomero84 I was looking at the implementation of `requiredDoceOrgFlag`. If you add property `required: true` you should be able to remove the `as Org` assertion.

Also, SfCommand `Flags` does export a `custom` method. I was curious why you are using OclifFlags version of that method.

```
export const requiredDoceOrgFlag = OclifFlags.custom({
  char: 'c',
  summary: messages.getMessage('flags.targetDoceOrg.summary'),
  parse: async (input: string | undefined) => getOrgOrThrow(input),
  default: async () => getOrgOrThrow(),
  defaultHelp: async () => (await getOrgOrThrow())?.getUsername(),
});
```

Reply:
Hi @peternhale - if we set it as required then the user will alway have to provide this flag right?  We want to give the user both possibilities: to specify an org using `-c` flag or defaulting to the `--target-devops-center` config variable.

Reply:
Regarding the custom flag, I based my implementation on this:
[https://github.com/salesforcecli/plugin-deploy-retrieve/blob/main/src/utils/flags.ts](https://github.com/salesforcecli/plugin-deploy-retrieve/blob/main/src/utils/flags.ts)

Also when I replace the OclifFlags flag by SfCommand flag I get an error: 

> Property 'custom' does not exist...

Am I missing something?

Reply:
Required does not mean the user has to provide a value, it just means that by the end of a the `await this.parse(<class>);` that there must be a value, either by use input or flag defaults.

Reply:
So this gives you and error?

```
export const requiredDoceOrgFlag = Flags.custom({
  char: 'c',
  summary: messages.getMessage('flags.targetDoceOrg.summary'),
  parse: async (input: string | undefined) => getOrgOrThrow(input),
  default: async () => getOrgOrThrow(),
  defaultHelp: async () => (await getOrgOrThrow())?.getUsername(),
});
```

Reply:
- I added the required attribute to requiredDoceOrgFlag and works as expected. Thanks.
- Yes, it gives me an error: 

> Property 'custom' does not exist on type '{ boolean: <T = boolean>(options?: Partial<BooleanFlag<T>> | undefined) => BooleanFlag<T>; directory: Definition<string, { exists?: boolean | undefined; }>; ... 10 more ...; optionalOrg: Definition

Reply:
@EstebanRomero84 check the import of Flags. Should be from `@salesforce/sf-plugins-core`

Reply:
As discussed in DM, I will update this flag once we upgrade to @salesforce/sf-plugin-core v2.